### PR TITLE
fix featureGate docs to use the new top level field in Cluster config

### PR DESCRIPTION
**Pull request by yashbhutwala**

### CODE REVIEW

**Comment by tao12345666333** [comment]
fixes #1572

File: site/content/docs/user/quick-start.md [hunk 1]
```diff
@@ -330,36 +330,8 @@ Feature gates are a set of key=value pairs that describe alpha or experimental f
...
-- |
-  kind: KubeletConfiguration
-  featureGates:
-    FeatureGateName: true
-- |
-  kind: KubeProxyConfiguration
-  featureGates:
-    FeatureGateName: true
+featureGates:
+ FeatureGateName: true
```
Comment (aojea) [inline site/content/docs/user/quick-start.md:334]: indent needs another space, right?

Comment (yashbhutwala) [reply site/content/docs/user/quick-start.md]: yes sir, you are correct, sorry about that 😛

File: site/content/docs/user/quick-start.md [hunk 2]
```diff
@@ -330,36 +330,8 @@ Feature gates are a set of key=value pairs that describe alpha or experimental f
...
-      "feature-gates": "FeatureGateName=true"
-- |
-  kind: KubeletConfiguration
-  featureGates:
-    FeatureGateName: true
-- |
-  kind: KubeProxyConfiguration
-  featureGates:
-    FeatureGateName: true
+featureGates:
```
Comment (aojea) [inline site/content/docs/user/quick-start.md:333]: this is the config used in the GA jobs
```yaml
# config for 1 control plane node and 2 workers (necessary for conformance)
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
networking:
  ipFamily: ipv4
nodes:
- role: control-plane
- role: worker
- role: worker
featureGates: {"AllAlpha":false,"AllBeta":false,"RotateKubeletClientCertificate":true}
```

I have doubts now, better ask @BenTheElder directly :sweat_smile:

Comment (yashbhutwala) [reply site/content/docs/user/quick-start.md:333]: @aojea as far as I see, that's the same data structure expressed in two different ways. (just key: value)

Comment (aojea) [reply site/content/docs/user/quick-start.md:333]: that's my understanding, but I prefer to defer when I'm not sure 😉

Comment (BenTheElder) [reply site/content/docs/user/quick-start.md:333]: this is just yaml differences.

**Comment by BenTheElder** [comment]
/lgtm
/approve
thanks!

**Pull request merged**